### PR TITLE
feat: Add BoxPlot demo

### PR DESCRIPTION
**Pull request by id2359**
I've added a markdown file (080_boxplot.md)  containing an example of Zakodium's BoxPlot component, overlaid with a scatterplot of some normally distributed data with added jitter ( regenerated dynamically each time.) 

The box plot demo is in the boxplot subdirectory.

I've added brief listings of the mandatory and optional properties for the component.

### CODE REVIEW

File: docs/boxplot/BoxPlot.demo.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import {
+  AnnotationBoxPlotProps,
+  Plot,
+  Annotations,
+  Annotation,
+  SeriesPoint,
+} from 'react-plot';
+import { ScatterSeries, Axis } from 'react-plot';
+
+import random from 'random/dist/cjs';
```
Comment (stropitek) [inline docs/boxplot/BoxPlot.demo.tsx:11]:
```suggestion
import random from 'random';
```

**Review by stropitek — COMMENTED** [review]
The code sandbox "plugin" does not allow yet for third-party dependencies to be used in demos

I can have a look on how to have them added to dependencies automatically

File: docs/boxplot/BoxPlot.demo.tsx (hunk 1 above)
Comment (lpatiny) [reply docs/boxplot/BoxPlot.demo.tsx:11]: If you want seeded random numbers please use our package: https://github.com/mljs/xsadd

Comment (targos) [reply docs/boxplot/BoxPlot.demo.tsx:11]: xsadd doesn't provide a way to generate numbers using a normal distribution

File: docs/boxplot/BoxPlot.demo.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import { median, quartiles, max, min } from 'ml-stat/array';
+
+export default function BoxPlotExample() {
+  // dynamically generate some normally distributed data about the arbritary value of 300
+  const generator = random.normal(300, 100);
+  const jitter = random.normal(0, 0.1);
```
Comment (lpatiny) [inline docs/boxplot/BoxPlot.demo.tsx:18]: Why you didn't use createRandomArray with a 'normal' distribution and a 'seed' from ml-spectra-processing as suggested in the issue ?

If something was missing you can always make a PR, it is one of our projects.

**Review by lpatiny — CHANGES_REQUESTED** [review]

**Comment by id2359** [comment]
Ok sorry about that, to be honest I was going off what was said in Telegram,.. I'll fix up.





On Monday, 23 May 2022, 10:54:55 pm AWST, Luc Patiny ***@***.***> wrote: 







@lpatiny requested changes on this pull request.

________________________________

In docs/boxplot/BoxPlot.demo.tsx:
> @@ -0,0 +1,55 @@
+import React from 'react';
+import {
+  AnnotationBoxPlotProps,
+  Plot,
+  Annotations,
+  Annotation,
+  SeriesPoint,
+} from 'react-plot';
+import { ScatterSeries, Axis } from 'react-plot';
+
+import random from 'random/dist/cjs';
+
+import { median, quartiles, max, min } from 'ml-stat/array';

For boxPlot you could use this method: https://github.com/mljs/spectra-processing/blob/master/src/x/xBoxPlot.ts as suggested in the issue.

ml-spectra-processing is a typescript project that is often a dependency of our projects.

________________________________

In docs/boxplot/BoxPlot.demo.tsx:
> +  AnnotationBoxPlotProps,
+  Plot,
+  Annotations,
+  Annotation,
+  SeriesPoint,
+} from 'react-plot';
+import { ScatterSeries, Axis } from 'react-plot';
+
+import random from 'random/dist/cjs';
+
+import { median, quartiles, max, min } from 'ml-stat/array';
+
+export default function BoxPlotExample() {
+  // dynamically generate some normally distributed data about the arbritary value of 300
+  const generator = random.normal(300, 100);
+  const jitter = random.normal(0, 0.1);

Why you didn't use createRandomArray with a 'normal' distribution and a 'seed' from ml-spectra-processing as suggested in the issue ?

If something was missing you can always make a PR, it is one of our projects.

File: docs/boxplot/BoxPlot.demo.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  Plot,
+  Annotations,
+  Annotation,
+  SeriesPoint,
+} from 'react-plot';
+import { ScatterSeries, Axis } from 'react-plot';
+
+import random from 'random/dist/cjs';
+
+import { median, quartiles, max, min } from 'ml-stat/array';
```
[the remark this answers is not in the document]
Comment (id2359) [reply docs/boxplot/BoxPlot.demo.tsx:13]: ok will do

File: docs/boxplot/BoxPlot.demo.tsx (hunk 2 above)
Comment (id2359) [reply docs/boxplot/BoxPlot.demo.tsx:18]: I was focussing on the Telegram, sorry- will try what you've hinted.

**Comment by id2359** [comment]
I'll flip the q1 q3 when new version of react-plot available?

**Comment by stropitek** [comment]
@id2359 if you reply to github issues by email, please make sure to remove all the email's content first. Otherwise it looks like [this](https://github.com/zakodium-oss/react-plot-docs/pull/22#issuecomment-1135288440). Thanks!

**Comment by lpatiny** [comment]
@id2359 The preview is currently crashing. Any idea why ?

image

**Comment by lpatiny** [comment]
In SandBox the page still crash. I don't know if it is easy to solve.

image

**Pull request merged**